tests/mcproto/types/test_identifier.py

```diff
@@ -0,0 +1,55 @@
+from mcproto.types.identifier import Identifier
+from tests.helpers import gen_serializable_test
+from mcproto.types.nbt import StringNBT
+
+
+gen_serializable_test(
+    context=globals(),
+    cls=Identifier,
+    fields=[("namespace", str), ("path", str)],
+    serialize_deserialize=[
+        (("minecraft", "stone"), b"\x0fminecraft:stone"),
+        (("minecraft", "stone_brick"), b"\x15minecraft:stone_brick"),
+        (("minecraft:stone_brick_slab", None), b"\x1aminecraft:stone_brick_slab"),
+        (("stone", None), b"\x0fminecraft:stone"),
+        (
+            ("minecraft.beta", "textures/gui/advancements/backgrounds/end.png"),
+            b"\x3cminecraft.beta:textures/gui/advancements/backgrounds/end.png",
+        ),
+    ],
+    validation_fail=[
+        (("minecr*ft", "stone_brick_slab_top"), ValueError),  # Invalid namespace
+        (("minecraft", "stone_brick_slab_t@p"), ValueError),  # Invalid path
+        (("", "something"), ValueError),  # Empty namespace
+        (("minecraft", ""), ValueError),  # Empty path
+        (("minecraft", "a" * 32767), ValueError),  # Too long
+    ],
+)
+
+
+def test__identifier_convert_from_tag():
+    """Assert that any # in the namespace is removed."""
+    assert Identifier("#minecraft:stone").namespace == "minecraft"
+    assert Identifier("#minecraft:stone").path == "stone"
+    assert Identifier("#minecraft:stone").serialize() == b"\x0fminecraft:stone"
```

Comment:
It might be worth it to test what happens if the `#` is somewhere in the middle/end.

Reply:
It will just fail the regex check, I added this to allow writing a # before a tag name (they appear that way in the F3 menu)